script.js

```diff
@@ -1,4 +1,326 @@
+var myScore = 0;
+var computerScore = 0;
+var numberOfDraws = 0;
+var totalNumberofRounds = myScore + computerScore + numberOfDraws;
+var userWinningPercentage = (myScore / totalNumberofRounds) * 100;
+var computerWinningPercentage = (computerScore / totalNumberofRounds) * 100;
+var currentGameMode = `waiting for user name`;
+var userName = ``;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var myOutputValue = ``;
+  //do this if awaiting user to input his/her name
+  if (currentGameMode == `waiting for user name`) {
+    userName = input;
+    currentGameMode = `Select game mode`;
```

Comment:
Instead of referencing string constants  (i.e. "select game mode"), best practice to store them in a global variable. Why? <-- becos using strings open us up to human error like mispelling the words or adding extra spaces whenever we reuse the string constant (i.e in line 18). These errors will cause your code not to run the way you expect it to. 

On the other hand, if you stored it as a variable, vscode will suggest the variable to you as you type, and the ability to choose from these suggestions will remove room for human errors.

To implement, it would look like this: `var MODE_SELECT_GAME = "MODE_SELECT_GAME".`. Note that the variable name is in screaming snake case. This is the casing convention for variables that are storing string constants. The difference in casing just helps us visually separate the normal variables from those that are storing string constants.